web/components/FcDrawerRequestStudyBulkEdit.vue

```diff
@@ -51,13 +61,26 @@ export default {
   beforeDestroy() {
     this.setLocationsIndicesDeselected([]);
   },
+  beforeRouteLeave(to, from, next) {
+    if (this.leaveConfirmed) {
+      next();
+    } else {
+      this.nextRoute = to;
+      this.showConfirmLeave = true;
+    }
+  },
   methods: {
-    actionCancel() {
-      this.$refs.nav.actionConfirmLeave();
+    actionLeave() {
+      this.leaveConfirmed = true;
+      this.$router.push(this.nextRoute);
+    },
+    actionNavigateBack(leaveConfirmed) {
```

Comment:
can we make the parameter `leaveConfirmed = false`?  this way we can guarantee at least that this takes a boolean value.  (same for other two `actionNavigateBack` component methods in edit non-bulk and create flows.)